packages/runner-ct/src/iframe/iframes.tsx

```diff
@@ -22,134 +21,133 @@ export function getSpecUrl ({ namespace, spec }, prefix = '') {
 interface IFramesProps {
   state: State
   eventManager: typeof eventManager
-  config: any
+  config: Cypress.RuntimeConfigOptions
 }
 
-@observer
-export default class Iframes extends Component<IFramesProps> {
-  _disposers = []
-  containerRef = null
-  autIframe: AutIframe
-  iframeModel: IframeModel
-
-  render () {
-    const { viewportHeight, viewportWidth, scriptError, scale, screenshotting } = this.props.state
-
-    return (
-      <div
-        style={{
-          display: this.props.state.screenshotting ? 'inherit' : 'grid',
-        }}
-        className={cs('iframes-ct-container', {
-          'has-error': !!scriptError,
-          'iframes-ct-container-screenshotting': screenshotting,
-        })}
-      >
-        <div
-          ref={(container) => this.containerRef = container}
-          className={
-            cs('size-container', {
-              [styles.noSpecAut]: !this.props.state.spec,
-            })
-          }
-          style={{
-            height: viewportHeight,
-            width: viewportWidth,
-            transform: `scale(${screenshotting ? 1 : scale})`,
-          }}
+export const Iframes: React.FC<IFramesProps> = namedObserver('Iframes', (props) => {
```

Comment:
I know you don't like destructuring props, but you really should do it. It would clean up this code quite a bit.

Reply:
Sure. I am not entirely sure how destructing makes this cleaner, though. Are you suggesting just to do something like:

```js
export const Iframes: React.FC<IFramesProps> = namedObserver('Iframes', ({
  state,
  eventManager
  // ... etc ...
}) => { 
```

?

Happy to make this change, I'd like to know what makes this cleaner/better than the existing style, though - I've seen the word "convention" tossed around, but no-one seems to be able to explain why it's objectively superior.

I don't have a strong opinion about this, it's just a coding style, but I don't want to do something without understanding why it's an improvement.